File: frontend/src/pages/MajorsView.js
import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router-dom';
import majorsData from '../majors.json';
import favoriteIcon from '../assets/favorite.png';
import './MajorsView.css';

function MajorsView() {
  const { id } = useParams();
  const major = majorsData.find(m => m.pk === parseInt(id));

  const [isFavorite, setIsFavorite] = useState(false);

  // ✅ Save to search history when a major is viewed (max 10 items)
  useEffect(() => {
    if (major) {
      let history = JSON.parse(localStorage.getItem("searchHistory")) || [];

      // Add new search to the front
      history = [major.fields.name, ...history.filter(item => item !== major.fields.name)];

      // Keep only the last 10
      history = history.slice(0, 10);

      localStorage.setItem("searchHistory", JSON.stringify(history));
    }
  }, [major]);

  useEffect(() => {
    const favorites = JSON.parse(localStorage.getItem('favorites')) || [];
    setIsFavorite(favorites.includes(major?.pk));
  }, [major]);

  const toggleFavorite = () => {
    const favorites = JSON.parse(localStorage.getItem('favorites')) || [];
    if (isFavorite) {
      // Remove from favorites
      const updated = favorites.filter(favId => favId !== major.pk);
      localStorage.setItem('favorites', JSON.stringify(updated));
      setIsFavorite(false);
    } else {
      // Add to favorites
      favorites.push(major.pk);
      localStorage.setItem('favorites', JSON.stringify(favorites));
      setIsFavorite(true);
    }
  };

  if (!major) return <p>Major not found!</p>;

  const employmentRate = major.fields.employment_rate;
  let circleColor = 'red';
  if (employmentRate >= 80) circleColor = 'green';
  else if (employmentRate >= 50) circleColor = 'orange';

  return (
    <div className="majors-view-container">
      <div className="info-box">
        <h2>{major.fields.name}</h2>
        <p>{major.fields.about}</p>
        <p><strong>Career Prospects:</strong> {major.fields.career_prospects}</p>
        <p><strong>Universities:</strong> {major.fields.universities}</p>
        <p><strong>Entry Level Salary:</strong> ${major.fields.entry_level_salary}</p>
        <p><strong>Mid Level Salary:</strong> ${major.fields.mid_level_salary}</p>
        <p><strong>Senior Level Salary:</strong> ${major.fields.senior_level_salary}</p>

        <div className="employment-circle" style={{ borderColor: circleColor }}>
          <span>{employmentRate}%</span>
        </div>

        <button className="favorite-btn" onClick={toggleFavorite}>
          <img src={favoriteIcon} alt="Favorite" className={isFavorite ? 'active' : ''} />
        </button>
      </div>
    </div>
  );
}

export default MajorsView;

File: backend/routes/majorsview.py
# # majors/views.py
# from django.shortcuts import render, get_object_or_404, redirect
# from .models import Major

# def major_detail(request, pk):
#     major = get_object_or_404(Major, pk=pk)

#     # --- Track search history (last 10) ---
#     search_history = request.session.get('search_history', [])
#     if major.name in search_history:
#         search_history.remove(major.name)
#     search_history.insert(0, major.name)
#     search_history = search_history[:10]
#     request.session['search_history'] = search_history

#     # --- Handle favorites ---
#     favorites = request.session.get('favorites', [])
#     is_favorite = major.pk in favorites

#     if request.method == 'POST' and 'toggle_favorite' in request.POST:
#         if is_favorite:
#             favorites.remove(major.pk)
#             is_favorite = False
#         else:
#             favorites.append(major.pk)
#             is_favorite = True
#         request.session['favorites'] = favorites
#         return redirect('major_detail', pk=major.pk)

#     # --- Employment rate with safe default ---
#     employment_rate = major.employment_rate if major.employment_rate is not None else 0

#     if employment_rate >= 80:
#         circle_color = 'green'
#     elif employment_rate >= 50:
#         circle_color = 'orange'
#     else:
#         circle_color = 'red'

#     context = {
#         'major': major,
#         'is_favorite': is_favorite,
#         'employment_rate': employment_rate,
#         'circle_color': circle_color,
#     }

#     return render(request, 'majors/majorsview.html', context)
